napari/_app_model/_tests/test_viewer_toggler.py

```diff
@@ -15,7 +15,13 @@ def test_viewer_toggler():
     app = get_app()
     app.register_action(action)
 
-    with app.injection_store.register(providers={Viewer: lambda: viewer}):
+    # Injection required as there is no current viewer, use a high weight (100)
+    # so this provider is used over `_provide_viewer`, which would raise an error
+    with app.injection_store.register(
+        providers=[
+            (lambda: viewer, Viewer, 100),
+        ]
+    ):
```

Comment:
There is no current viewer so `_provide_viewer` would not work. It would work if I used `make_napari_viewer` first here but I thought it did not need the unnecessary overhead as that runs a lot of things we don't need (e.g., `initialize_plugins`). Happy to change.

Reply:
But injection exists ealier, I ask why it is changed here, why weight is required.

Reply:
Yeah this was tricky.

On `app` init the `_provide_qt_viewer` is registered. You can register more than one provider per type hint, so here I register a 2nd viewer provider but weight it highly so that it is used in preference to `_provide_qt_viewer`. This is because `_provide_qt_viewer` will not work as there is no current provider. 

Not sure what you mean by 'injection exists earlier' - but I need to register the provider before executing the commands.

Reply:
Yes! That is why I add the `weight` so the 2nd provider is used in priority